modules/clients/src/test/java/org/apache/ignite/internal/jdbc2/JdbcClobTest.java

```diff
@@ -0,0 +1,448 @@
+
+        assertThrows(null, () -> clob.getCharacterStream(1, 11), SQLException.class, null);
+
+        Reader cStream = clob.getCharacterStream(1, 10);
+        String res = IOUtils.toString(cStream);
+        assertEquals("1234567890", res);
+
+        cStream = clob.getCharacterStream(1, 1);
+        res = IOUtils.toString(cStream);
+        assertEquals("1", res);
+
+        cStream = clob.getCharacterStream(10, 1);
+        res = IOUtils.toString(cStream);
+        assertEquals("0", res);
+
+        cStream = clob.getCharacterStream(3, 5);
+        res = IOUtils.toString(cStream);
+        assertEquals("34567", res);
+
+        cStream = clob.getCharacterStream(3, 0);
+        res = IOUtils.toString(cStream);
+        assertEquals("", res);
+
+        clob.free();
+        assertThrows(null, () -> clob.getCharacterStream(1, 1), SQLException.class, ERROR_CLOB_FREE);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testGetAsciiStream() throws Exception {
+        JdbcClob clob = new JdbcClob("1234567890");
+        byte[] bytes = IOUtils.toByteArray(clob.getAsciiStream());
+        Assert.assertArrayEquals("1234567890".getBytes(UTF_8), bytes);
+
+        clob.free();
+        assertThrows(null, clob::getAsciiStream, SQLException.class, ERROR_CLOB_FREE);
+
+        Clob emptyClob = new JdbcClob("");
+        bytes = IOUtils.toByteArray(emptyClob.getAsciiStream());
+        Assert.assertArrayEquals("".getBytes(UTF_8), bytes);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testGetAsciiStreamForNonAsciiDataBufferedRead() throws Exception {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < 10000; i++) {
+            sb.append("aa©😀");
+        }
+
+        Clob clob = new JdbcClob(sb.toString());
+
+        InputStream stream = clob.getAsciiStream();
+
+        assertThrows(null, () -> stream.read(null, 0, 1), NullPointerException.class, null);
+
+        assertThrows(null, () -> stream.read(new byte[10], -1, 5), IndexOutOfBoundsException.class, null);
+
+        assertThrows(null, () -> stream.read(new byte[10], 5, -1), IndexOutOfBoundsException.class, null);
+
+        assertThrows(null, () -> stream.read(new byte[10], 11, 1), IndexOutOfBoundsException.class, null);
+
+        assertThrows(null, () -> stream.read(new byte[10], 5, 6), IndexOutOfBoundsException.class, null);
+
+        assertEquals(0, stream.read(new byte[10], 5, 0));
+
+        byte[] bytes = IOUtils.toByteArray(stream);
+
+        String reencoded = new String(bytes, UTF_8);
+
+        assertEquals(clob.getSubString(1, (int)clob.length()), reencoded);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testGetAsciiStreamForNonAsciiDataReadByByte() throws Exception {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < 10000; i++) {
```

Comment:
do we really need 10000 copies of the string?
Let's help CI and keep 10 copies?

Reply:
The idea was to cover all variants in the JdbcClob.Utf8EncodedStringInputStream.encodeNextChunk() implementaion.

I reduce this to 3277 in one of two testcases (the minimum which produces the malformed surrogate).

And change to 10 in another one.